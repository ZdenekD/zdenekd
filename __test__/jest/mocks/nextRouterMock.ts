const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        asPath: '/page/subpage/something-else',
        pathname: '/',
        route: '/',
        locale: 'cs',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
