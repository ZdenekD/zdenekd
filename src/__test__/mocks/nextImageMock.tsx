const nextImageMock = jest.mock('next/image', () => function Image() {
    return <img src="" alt="" />;
});

export default nextImageMock;
