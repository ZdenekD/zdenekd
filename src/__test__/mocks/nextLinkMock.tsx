const nextLinkMock = jest.mock('next/link', () => function Link() {
    return <a href="/">link</a>;
});

export default nextLinkMock;
