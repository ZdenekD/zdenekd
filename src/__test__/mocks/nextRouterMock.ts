const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'en',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
