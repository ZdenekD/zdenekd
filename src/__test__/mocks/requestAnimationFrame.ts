import react from 'react';

window.requestAnimationFrame = () => 0;

export default react;
