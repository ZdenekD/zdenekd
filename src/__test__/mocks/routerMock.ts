const routerMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'cs',
        locales: ['cs', 'en'],
    })),
}));

export default routerMock;
