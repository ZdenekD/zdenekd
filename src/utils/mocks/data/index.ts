import MockDataParagraph from './content/UI/paragraph';

export {MockDataParagraph};
