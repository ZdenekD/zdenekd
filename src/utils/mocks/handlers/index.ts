import handler from './handler';

const handlers = [...handler];

export default handlers;
