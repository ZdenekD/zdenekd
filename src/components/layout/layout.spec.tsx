import {shallow, ShallowWrapper} from 'enzyme';
import findComponent from '../../__test__/utils/helpers';
import '../../__test__/mocks/nextRouterMock';
import Layout from '.';

describe('Layout', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<Layout>Layout content</Layout>);
    });

    it('match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const component = findComponent(wrapper, 'component-layout');

        expect(component.exists()).toBe(true);
    });
});
