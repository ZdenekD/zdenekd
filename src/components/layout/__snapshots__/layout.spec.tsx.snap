// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout match snapshot 1`] = `
<main class="main "
      data-test="component-layout"
>
  <header class="header"
          data-test="component-header"
  >
    <a href="/"
       class="link "
       aria-label="Odkaz na domovskou stránku"
    >
      <div class="wrapper ">
        <svg data-filename="src/assets/images/logo.svg">
        </svg>
        <i class="shadow"
           style="opacity:1"
        >
        </i>
      </div>
    </a>
    <button type="button"
            class="button closed "
            data-test="component-menu"
            aria-label="Otevřít navigaci"
    >
      <svg class="icon"
           viewbox="0 0 100 100"
      >
        <path d="M30,40 L70,40 Z">
        </path>
        <path d="M30,50 L70,50 Z">
        </path>
        <path d="M30,60 L70,60 Z">
        </path>
      </svg>
    </button>
  </header>
  <section class="section"
           data-test="component-section"
           style="opacity:0"
  >
    <noscript>
      <strong>
        Nemáte zapnutý Javascript
      </strong>
      <p>
        Některé součásti stránek Vám nemusí fungovat správně.
        <br>
        Zde jsou
        <a href="https://www.enable-javascript.com/cz/"
           target="_blank"
           rel="noreferrer noopener"
        >
          instrukce
        </a>
        jak povolit JavaScript ve Vašem webovém prohlížeči.
      </p>
    </noscript>
    Content
  </section>
  <footer class="footer closed"
          data-test="component-footer"
          style="opacity:0"
  >
    <code class="code">
      ♣ MMXXI Vyrobeno pomocí &lt;kouzel&gt; a {zaklínadel}
    </code>
    <div class="linkWrapper">
      <a href="mailto:ditezdenek@pm.me?subject=♣ Zpráva přímo do schránky ♣"
         title="Zpráva přímo do schránky"
         class="link"
      >
        <svg class="icon icon-tabler icon-tabler-message"
             width="28"
             height="28"
             viewbox="0 0 24 24"
             stroke-width="1"
             stroke="currentColor"
             fill="none"
             stroke-linecap="round"
             stroke-linejoin="round"
        >
          <path stroke="none"
                d="M0 0h24v24H0z"
                fill="none"
          >
          </path>
          <path d="M4 21v-13a3 3 0 0 1 3 -3h10a3 3 0 0 1 3 3v6a3 3 0 0 1 -3 3h-9l-4 4">
          </path>
          <line x1="8"
                y1="9"
                x2="16"
                y2="9"
          >
          </line>
          <line x1="8"
                y1="13"
                x2="14"
                y2="13"
          >
          </line>
        </svg>
      </a>
      <a href="https://github.com/ZdenekD"
         title="Github"
         class="link"
         target="_blank"
         rel="noreferrer noopener"
      >
        <svg class="icon icon-tabler icon-tabler-brand-github"
             width="28"
             height="28"
             viewbox="0 0 24 24"
             stroke-width="1"
             stroke="currentColor"
             fill="none"
             stroke-linecap="round"
             stroke-linejoin="round"
        >
          <path stroke="none"
                d="M0 0h24v24H0z"
                fill="none"
          >
          </path>
          <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5">
          </path>
        </svg>
      </a>
    </div>
    <div class="keys"
         data-test="component-keys"
         style="opacity:1"
    >
      <button type="button"
              class="key "
              aria-label="Předchozí stránka"
              data-test="component-key"
      >
        <i class="icon up">
        </i>
      </button>
      <div class="key">
        <button type="button"
                class="key disabled"
                aria-label="Předchozí projekt"
                data-test="component-key"
                disabled
        >
          <i class="icon left">
          </i>
        </button>
        <button type="button"
                class="key "
                aria-label="Následující stránka"
                data-test="component-key"
        >
          <i class="icon down">
          </i>
        </button>
        <button type="button"
                class="key disabled"
                aria-label="Následující projekt"
                data-test="component-key"
                disabled
        >
          <i class="icon right">
          </i>
        </button>
      </div>
    </div>
    <span class="scroll"
          style="opacity:1"
    >
      Scroll
      <i class="dot"
         style="opacity:1;transform:none"
      >
      </i>
    </span>
  </footer>
  <div class="curtain"
       data-test="component-curtain"
       style="background:var(--color-yellow-lighten-4);transform:none"
  >
  </div>
</main>
<aside class="aside"
       data-test="component-aside"
       style="transform:translateY(-100%) translateZ(0)"
>
  <nav class="nav"
       data-test="component-nav"
  >
    <ul class="list"
        style="opacity:0"
    >
      <li class="item"
          data-test="component-item"
          style="opacity:0;transform:translateY(-6vh) translateZ(0)"
      >
        <a href="/"
           class="anchor link "
           data-test="component-anchor"
        >
          Úvodní stránka
        </a>
      </li>
      <li class="item"
          data-test="component-item"
          style="opacity:0;transform:translateY(-6vh) translateZ(0)"
      >
        <a href="/projects"
           class="anchor link "
           data-test="component-anchor"
        >
          Projekty
        </a>
      </li>
      <li class="item"
          data-test="component-item"
          style="opacity:0;transform:translateY(-6vh) translateZ(0)"
      >
        <a href="/contact"
           class="anchor link "
           data-test="component-anchor"
        >
          Kontakt
        </a>
      </li>
    </ul>
  </nav>
</aside>
<i class="outerCursor"
   data-test="component-outer-cursor"
>
</i>
<i class="innerCursor "
   data-test="component-inner-cursor"
>
</i>
`;
